test/files/python_whitespace.expected

```diff
@@ -4,13 +4,13 @@ Failure in python_whitespace.py:
     {{^}} four
 
   which failed to match line stdout:4:
-    four
+       four
```

Comment:
This now prints leading whitespace, which we typically ignore, but in cases like this we don't, explicitly.

I've also needed leading whitespace a few times, e.g. when checking that the `^` carets are aligned correctly, so this seems better.

We could also strip leading whitespace for matching expressions, or show a leading `|` to mark where the line starts?

Reply:
The indent seems alright as long as the structure of the output suggests that the indented part is actual line, and the non-indented part is just metadata.

It seems inconsistent though, `four` only has 4+3 spaces, while the `one` from the context has 4+4 spaces.

Reply:
>It seems inconsistent though, four only has 4+3 spaces, while the one from the context has 4+4 spaces.

That's because it actually is that way in the test file. The output just shows [what is there](https://github.com/ridiculousfish/littlecheck/blob/7d232ee79c6a03593d4c280f7a422afee60e9c17/test/files/python_whitespace.py#L13). (which is inconsistent and not necessary, but it's not caused by this patch)

>The indent seems alright as long as the structure of the output suggests that the indented part is the actual line, and the non-indented part is just metadata.

That's why I'm asking if we add a delimiter. Because output like

```
  Context:
    a
      b
```

doesn't really make clear that there are *2 spaces* before the `b`, because it's 2+4 already.

Do we change it to

```
  Context:
  |a
  |  b
```

so the line starts right after the `|`? (note that this will only really work in a monospace font, which is a given in a terminal, but not e.g. on Github Actions in a browser)

Reply:
I see, I was looking at the wrong lines in python_whitespace.py.

Either option is fine by me. I think it's okay like this, without a delimiter.

>  not e.g. on Github Actions in a browser

https://github.com/fish-shell/fish-shell/runs/1402741560?check_suite_focus=true looks like monospace.
Anyway, even without the `|` it will look weird in a non-monospace font, I think we can ignore that.